Source/Shared/BPXCTestFile.m

```diff
@@ -112,14 +112,29 @@ + (instancetype)BPXCTestFileFromDictionary:(NSDictionary *)dict
         return nil;
     }
     testHostPath = [testHostPath stringByReplacingOccurrencesOfString:TESTROOT withString:testRoot];
-
     NSString *testBundlePath = [dict objectForKey:@"TestBundlePath"];
     if (!testBundlePath) {
         BP_SET_ERROR(error, @"No 'TestBundlePath' found");
         return nil;
     }
-    testBundlePath = [testBundlePath stringByReplacingOccurrencesOfString:TESTHOST withString:testHostPath];
-
+    /*testBundlePath is expected to be "__TESTHOST__/PlugIns/.."
+      temporal workaround for bug: 34135468, Xcode9beta5 Inconsistent TestBundlePath With "xcodebuild build-for-testing” Command
+      when it is "__TESTROOT__"
+      testHostPath path is: /Users/xzhang3/Desktop/Documents/Git/bluepill/build/Build/Products/Debug-iphonesimulator/bluepill/
+      testBundlePath is: __TESTROOT__/Debug-iphonesimulator/BPSampleApp.app/PlugIns/BPSampleAppHangingTests.xctest
+      expected path is: /Users/xzhang3/Desktop/Documents/Git/bluepill/build/Build/Products/Debug-iphonesimulator/BPSampleApp.app/
+      testHost path is: __TESTROOT__/Debug-iphonesimulator/BPSampleApp.app
+     */
+    if ([testBundlePath rangeOfString:TESTHOST].location != NSNotFound) {
+        testBundlePath = [testBundlePath stringByReplacingOccurrencesOfString:TESTHOST withString:testHostPath];
+    } else if ([testBundlePath rangeOfString:TESTROOT].location != NSNotFound) {
+        testHostPath = [testHostPath stringByDeletingLastPathComponent]; //remove /bluepill
+        NSString *temp = [testHostPath stringByDeletingLastPathComponent]; //remove /iphonesimulator
+        testHostPath = [testHostPath stringByAppendingString:@"/BPSampleApp.app"];
```

Comment:
what is this?

Reply:
Temporal workaround for bug: 34135468, Xcode9beta5 Inconsistent TestBundlePath With "xcodebuild build-for-testing” Command

Reply:
Why d you need to hardcode ` testHostPath = [testHostPath stringByAppendingString:@"/BPSampleApp.app"];` ?

Reply:
No, we should not hardcode it. Updated in the latest commit